test/fakeOrigin/endpoint/endpoint.go

```diff
@@ -0,0 +1,216 @@
+const DefaultConfigFile = "config.json"
+
+// DefaultOutputDirectory is the default output directory for generated content
+const DefaultOutputDirectory = "./out"
+
+// DefaultHTTPSKeyFile is the default path to the SSL key
+const DefaultHTTPSKeyFile = "server.key"
+
+// DefaultHTTPSCertFile is the default path to the SSL certificate
+const DefaultHTTPSCertFile = "server.cert"
+
+// ServerInfo contains relevant info for serving content
+type ServerInfo struct {
+	HTTPListeningPort  int    `json:"http_port"`
+	HTTPSListeningPort int    `json:"https_port"`
+	SSLCert            string `json:"ssl_cert"`
+	SSLKey             string `json:"ssl_key"`
+	BindingAddress     string `json:"binding_address"`
+	CrossdomainFile    string `json:"crossdomain_xml_file"`
+}
+
+// Endpoint defines all kinds of endpoints to be served
+type Endpoint struct {
+	ID              string              `json:"id"`
+	DiskID          string              `json:"override_disk_id,omitempty"`
+	Source          string              `json:"source"`
+	OutputDirectory string              `json:"outputdir,omitifempty"`
+	EndpointType    Type                `json:"type"`
+	ManualCommand   []string            `json:"manual_command,omitempty"`
+	DefaultHeaders  map[string][]string `json:"default_headers,omitempty"`
+	NoCache         bool                `json:"no_cache,omitempty"`
+	ABRManifests    []string            `json:"abr_manifests,omitempty"`
+}
+
+// Config defines the application configuration
+type Config struct {
+	ServerConf ServerInfo `json:"server"`
+	Endpoints  []Endpoint `json:"endpoints"`
+}
+
+func contains(args []string, param string) bool {
+	for _, str := range args {
+		if strings.Contains(str, param) {
+			return true
+		}
+	}
+	return false
+}
+
+func replace(tokenDict map[string]string, input string) string {
+	for k, v := range tokenDict {
+		input = strings.Replace(input, k, v, -1)
+	}
+	return input
+}
+
+// DefaultConfig generates a basic default configuration
+func DefaultConfig() Config {
+	return Config{
+		ServerConf: ServerInfo{
+			HTTPListeningPort:  8080,
+			HTTPSListeningPort: 8443,
+			SSLCert:            DefaultHTTPSCertFile,
+			SSLKey:             DefaultHTTPSKeyFile,
+			BindingAddress:     "",
+			CrossdomainFile:    "./example/crossdomain.xml",
+		},
+		Endpoints: []Endpoint{
+			{
+				ID:              "SampleFile",
+				Source:          "./example/video/kelloggs.mp4",
+				OutputDirectory: DefaultOutputDirectory,
+				EndpointType:    Static,
+			},
+			{
+				ID:              "SampleDirectory",
+				Source:          "./example/video/",
+				OutputDirectory: DefaultOutputDirectory,
+				EndpointType:    Dir,
+			},
+		},
+	}
+}
+
+// WriteConfig saves a fakeOrigin config as a pretty-printed json file
+func WriteConfig(cfg Config, path string) error {
+	bts, err := json.MarshalIndent(cfg, "", "\t")
+	if err != nil {
+		return errors.New("marshalling JSON: " + err.Error())
+	}
+	if err = ioutil.WriteFile(path, bts, 0755); err != nil {
```

Comment:
This writes non-POSIX-compliant files - needs a terminating newline. I think if you just change the argument here from `bts` to `append(bts, '\n')` it should work.

Also, the file you're writing is executable, shouldn't the FileMode argument here be `0666`?

Reply:
or <math><mrow><mn>0644<mn></mrow></math>, actually lines up more with what you've got atm

Reply:
Resolved via commit